src/Servers/Kestrel/Core/src/KestrelServer.cs

```diff
@@ -57,4 +62,44 @@ public void Dispose()
     {
         _innerKestrelServer.Dispose();
     }
+
+    private sealed class SimpleHttpsConfigurationService : IHttpsConfigurationService
+    {
+        public bool IsInitialized => true;
+
+        public void Initialize(IHostEnvironment hostEnvironment, ILogger<KestrelServer> serverLogger, ILogger<HttpsConnectionMiddleware> httpsLogger)
+        {
+            // Already initialized
+        }
+
+        public void PopulateMultiplexedTransportFeatures(FeatureCollection features, ListenOptions listenOptions)
+        {
+            HttpsConfigurationService.PopulateMultiplexedTransportFeaturesWorker(features, listenOptions);
+        }
+
+        public ListenOptions UseHttpsWithDefaults(ListenOptions listenOptions)
+        {
+            return HttpsConfigurationService.UseHttpsWithDefaultsWorker(listenOptions);
+        }
+
+        public void ApplyHttpsConfiguration(
+            HttpsConnectionAdapterOptions httpsOptions,
+            EndpointConfig endpoint,
+            KestrelServerOptions serverOptions,
+            CertificateConfig? defaultCertificateConfig,
+            ConfigurationReader configurationReader)
+        {
+            throw new NotImplementedException(); // Not actually required by this impl
+        }
+
+        public ListenOptions UseHttpsWithSni(ListenOptions listenOptions, HttpsConnectionAdapterOptions httpsOptions, EndpointConfig endpoint)
+        {
+            throw new NotImplementedException(); // Not actually required by this impl
+        }
+
+        public CertificateAndConfig? LoadDefaultCertificate(ConfigurationReader configurationReader)
+        {
+            throw new NotImplementedException(); // Not actually required by this impl
+        }
+    }
```

Comment:
I feel like I'm missing something. This is saying that these methods aren't called when KestrelServer is newed up manually?

Reply:
Correct, `KestrelServer` and `KestrelServerImpl` only pass the `IHttpsConfigurationService` to `TransportManager` and `AddressBinder`, so they don't need the parts of the interface that are exclusively for `KestrelConfigurationLoader`.